### ISSUE TITLE
ImageProvider generated as InvalidType

### ISSUE THREAD

**Issue by yakou32** [rrousselGit/riverpod#3525]
**Bug description**
I'm using Flutter Riverpod 2.4.9 to generate an `ImageProvider` object (either FileImage, AssetImage, NetworkImage...), according to some variable.

The generator doesn't compile correctly, the `ImageProvider` type being replaced by `InvalidType`.

**To Reproduce**

Here is a simplified code to reproduce:
```dart
// test.dart

import 'package:flutter/material.dart';
import 'package:riverpod_annotation/riverpod_annotation.dart';

part 'test.g.dart';

@riverpod
ImageProvider testImage(TestImageRef ref) {
  return const AssetImage('some_valid_path');
}

```

**Expected behavior**
The generated file should look like 
   `final testImageProvider = AutoDisposeProvider<ImageProvider>.internal(...`
instead of
   `final testImageProvider = AutoDisposeProvider<InvalidType>.internal(...`

**Workaround**
At the moment, I replace the current type by hand in the generate file.

It is obviously a tedious task; best would be to fix the ImageProvider compatibility issue instead.

Is it because the object class name `ImageProvider` would interfere with any Riverpod logic?

**Comment by snapsl** [comment]
Have you tried Raw<ImageProvider>?

**Comment by yakou32** [comment]
Thanks @snapsl I was not aware of that [`Raw` type](https://riverpod.dev/docs/essentials/websockets_sync#disabling-conversion-of-streamsfutures-to-asyncvalue). 

However, it doesn't solve the issue.
Indeed, in the simplified example, there is no Stream nor Future that Raw would disable...

**Comment by rrousselGit** [comment]
This is a known issue with various types.
Only 3.0 fixes the issue, and the fix won't be added to 2.0

**Comment by yakou32** [comment]
That's great!

Thank you so much Rémi!
